### PR TITLE
Only include libc and num_threads if compiling on unix with local offset

### PR DESCRIPTION
**Pull request by MarcusGrass**
libc and num_threads are only used on linux with the `local_offset` feature so they can be made optional and only included then. 
At least it worked compiling for x86_64-pc-windows-gnu without errors and the dependencies were not included.

### CODE REVIEW

File: time/Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ default = ["std"]
 alloc = ["serde?/alloc"]
 formatting = ["dep:itoa", "std", "time-macros?/formatting"]
 large-dates = ["time-macros?/large-dates"]
-local-offset = ["std"]
+local-offset = ["std", "libc", "num_threads"]
```
Comment (jhpratt) [inline time/Cargo.toml:29]:
```suggestion
local-offset = ["std", "dep:libc", "dep:num_threads"]
```

This avoids introducing implicit feature flags.

File: time/src/macros.rs [hunk 2]
```diff
@@ -65,7 +65,6 @@ pub use time_macros::datetime;
 /// );
 /// # Ok::<_, time::Error>(())
 /// ```
-///
```
Comment (jhpratt) [inline time/src/macros.rs:68]: You can remove these changes. I know `rustfmt` is failing, but it's a bug upstream and not something that needs to be changed here.

**Review by jhpratt — CHANGES_REQUESTED** [review]
I would've sworn that this wasn't supported by Cargo, but it definitely works when I have this locally. So I see no reason to not do this! Just a minor change.

File: time/Cargo.toml (hunk 1 above)
Comment (MarcusGrass) [reply time/Cargo.toml]: Yeah there's so much stuff you can do that I don't know about, the dep thing is nice though, should have spotted it from the other imports!

Comment (jhpratt) [reply time/Cargo.toml]: No worries! It's relatively new, only about 7 months old on stable. I did add a note (mostly for myself) above `[dependencies]`, but you had no reason to look there.

**Comment by jhpratt** [comment]
Thanks again!

**Pull request merged**